src/role-access/containers/specific-access/specific-access-duration/specific-access-duration.component.ts

```diff
@@ -1,60 +1,197 @@
-import { Component, Input, OnDestroy, OnInit } from '@angular/core';
-import { FormControl, FormGroup, Validators } from '@angular/forms';
-import { PersonRole } from '@hmcts/rpx-xui-common-lib/lib/models/person.model';
-import { Store } from '@ngrx/store';
-import { Subscription } from 'rxjs';
-import { Observable } from 'rxjs/Observable';
-import { ERROR_MESSAGE, PERSON_ROLE } from '../../../constants';
-import { SpecificAccessNavigationEvent, SpecificAccessState } from '../../../models';
-import { SpecificAccessNavigation } from '../../../models/specific-access-navigation.interface';
+import { Component, Input, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { select, Store } from '@ngrx/store';
+import {
+  DurationTypeDescription,
+  Period,
+  SpecificAccessStateData,
+  SpecificAccessNavigation,
+  SpecificAccessNavigationEvent
+} from '../../../models';
+import { DurationType } from '../../../models/enums';
+import { DurationHelperService } from '../../../services/duration-helper.service';
```

Comment:
Minor comment - Getting service from direct file, could it not be put into respective index.ts?

Reply:
Have updated import